libavcodec/qsvenc.h

```diff
@@ -212,4 +249,73 @@ int ff_qsv_encode(AVCodecContext *avctx, QSVEncContext *q,
 
 int ff_qsv_enc_close(AVCodecContext *avctx, QSVEncContext *q);
 
+/**
+ * Verify a key that specifies a specific qsv parameter exists and its value
+ * is valid
+ */
+int ff_qsv_validate_params(char *key, char *value);
+
+/**
+ * validate master display values provided as input and check if they are within HDR
+ * range
+ */
+int ff_qsv_validate_display(char *value);
+
+/**
+ * Extract the display_color values once verified and store in an AVMasteringDisplayMetadata
+ * struct
+ */
+int ff_qsv_extract_display_values(char *value, AVMasteringDisplayMetadata *aVMasteringDisplayMetadata);
+
+/**
+ * Verify content light level is within acceptable HDR range
+ */
+int ff_qsv_validate_cll(char *value);
+
+/**
+ * Parse and extract values from a string and store them in an array as integers
+ */
+int ff_qsv_extract_values(char *input_value, int *output_values);
+
+/**
+ * validate display's RGB and white point color levels
+ * @param input_colors an array representation of the X and Y values of the color.
+ */
+int ff_qsv_validate_display_color_range(int *input_colors);
+
+/**
+ * validate display's luminousity
+ * @param input_luminousity an array representation of the min and max values of the display luminousity.
+ */
+
+int ff_qsv_validate_display_luminousity_range(int *input_luminousity);
+
+/**
+ * Obtain video range value from the acceptable strings
+ * @param value a predefined string defining the video color range as per ISO/IEC TR 23091-4.
+ */
+
+int ff_qsv_extract_range_value(char *value);
+
+/**
+ * Obtain source video light level and assign these values to AVContentLightMetadata reference
+ * @param value a predefined string defining the light level.
+ * @param light_meta AVContentLightMetadata reference.
+ */
+ int ff_qsv_get_content_light_level(char *value, AVContentLightMetadata *light_meta);
+
+/**
+ * convert AVCodecContext color range value to equivalent mfx color range
+ * @param AVCodecContext color range
+ */
+int ff_avcol_range_to_mfx_col_range(int color_range);
+
+/**
+* Extract a substring from a string from a specified index
+ * @param start_index, where the target substring offset begins
+ * @param input_string, pointer to the first character of the original long string
+ * @param ouput_string,  pointer to the first character of the string variable that will hold the result substring
+ */
+void ff_build_sub_string(char *input_string, char *ouput_string, int start_index);
+
```

Comment:
Please add static local functions without ff_ prefix in qsvenc.c if they are not used outside of qsvenc.c.  See http://ffmpeg.org/developer.html#toc-Naming-conventions for naming conventions in FFmpeg